### PR TITLE
refactor metamask + sequence buy flow

### LINKED ISSUE #335: update Metamask buy flow
use matcha.xyz for swap + internal marketplace for buying USDC-PoS

definitely for Metamask, open question for other wallet providers?

### LINKED ISSUE #342: Replace Uniswap with Zeroswap when a user is signed in with Sequence
https://glodollar.slack.com/archives/C04PNCYTT89/p1691509154052349

### PR DESCRIPTION
**Pull request by Syncretik**
> switched marketplace to Metamask portfolio and DEX to matcha
> resolves #335, can extend PR to also accommodate #342 if needed 

> one caveat is that i'm not sure MM portfolio or matcha support robust deeplinking :/

resolves #342, #335

### CODE REVIEW

**Comment by gglucass** [comment]
Thanks for the pro-activity @Syncretik - love to see it!

It's a little bit unclear to me why we'd swap out Metamask for Matcha though. Let's discuss in our 1-1 later today before merging.

File: utils.ts [hunk 1]
```diff
@@ -138,16 +139,20 @@ export const getUSDCToUSDGLOUniswapDeeplink = (
...
   } else {
     outputCurrency = getSmartContractAddress(polygon.id);
-    uniswapChain = "polygon";
+    swapChain = "polygon";
   }
 
-  return `https://app.uniswap.org/#/swap?inputCurrency=${inputCurrency}&outputCurrency=${outputCurrency}&exactAmount=${amount}&exactField=input&chain=${uniswapChain}`;
+  const outputUrl =
+    dex === "Matcha"
+      ? `https://matcha.xyz/tokens/${swapChain}/${outputCurrency}`
```
Comment (gglucass) [inline utils.ts:153]: Not sure this is deep linked enough?

Comment (Syncretik) [reply utils.ts:153]: I tried adding both input and output currencies like 

`https://matcha.xyz/tokens/polygon/0x2791bca1f2de4661ed88a30c99a7a9449aa84174/0x4f604735c1cf31399c6e711d5962b2b3e0225ad3` but it doesn't work :(

**Comment by gglucass** [comment]
@Syncretik link into Zeroswap doesn't work for me

**Comment by Syncretik** [comment]
@gglucass I didn't request rer-eview, I'm still working on it

**Comment by Syncretik** [comment]
https://www.loom.com/share/bade159ceb134eee8614605007b3a08d

think it's good to go?

**Comment by gglucass** [comment]
> https://www.loom.com/share/bade159ceb134eee8614605007b3a08d
> 
> think it's good to go?

The loom looks good, but the latest deployment still opens in an about:blank for me 🤔 ?

**Comment by Syncretik** [comment]
yes that was what the loom was highlighting, because we're now forcing switch to Polygon mainnet on Sequence connection everytime even locally since last week.

if you see `utils.ts` we are still only allowing Mumbai on development (ref. screenshot)
Screen Shot 2023-08-15 at 6 16 40 PM



so for testing purposes we have to manually now switch to mumbai for the link to work.

**Review by gglucass — APPROVED** [review]

**Review by har777 — APPROVED** [review]

**Pull request merged**